crates/core/tedge/src/cli/http/cli.rs

```diff
@@ -219,3 +255,26 @@ fn http_client(
     };
     Ok(builder.build()?)
 }
+
+impl Content {
+    pub fn length(&self) -> Option<usize> {
+        if let Some(content) = &self.arg2 {
+            Some(content.len())
+        } else if let Some(data) = &self.data {
+            Some(data.len())
+        } else if let Some(file) = &self.file {
+            Some(std::fs::metadata(file).ok()?.len().try_into().ok()?)
+        } else {
+            None
+        }
+    }
+
+    pub fn mime_type(&self) -> Option<String> {
+        let file = self.file.as_ref()?;
+        Some(
+            mime_guess::from_path(file)
+                .first_or_octet_stream()
```

Comment:
Would it be better to default to `None` rather than octet stream, so that we don't send the mime type header at all, rather than possibly sending the wrong one? This would even mess with the default logic at the `HttpAction` level, right?

Reply:
`"application/octet-stream"` is general enough to be safe.